qiskit_ibm_experiment/service/ibm_experiment_service.py

```diff
@@ -463,7 +464,10 @@ def experiment(
         """
         with map_api_error(f"Experiment {experiment_id} not found."):
             raw_data = self._api_client.experiment_get(experiment_id)
-        return self._api_to_experiment_data(json.loads(raw_data, cls=json_decoder))
```

Comment:
maybe just make `self._api_to_experiment_data()` return `ExperimentData` instead of a dict?